File: API/WoodTrack.Core/Models/User/UserUpsertModel.cs
using Microsoft.AspNetCore.Http;

namespace WoodTrack.Core.Models;

public class UserUpsertModel : BaseUpsertModel
{
    public string FirstName { get; set; } = default!;
    public string LastName { get; set; } = default!;
    public string UserName { get; set; } = default!;
    public string Email { get; set; } = default!;
    public string? OldPassword { get; set; } = default!;
    public string? NewPassword { get; set; } = default!;
    public string PhoneNumber { get; set; } = default!;
    public DateTime? BirthDate { get; set; }
    public Gender? Gender { get; set; }
    public string? ProfilePhoto { get; set; }
    public string? ProfilePhotoThumbnail { get; set; }
    public string Address { get; set; } = default!;
    public string? Description { get; set; }
    public string? LicenseNumber { get; set; }
    public int? YearsOfExperience { get; set; }
    public string? WorkingHours { get; set; }
    public string? Position { get; set; }
    public int? CountryId { get; set; }
    public bool IsClient { get; set; }
    public bool IsEmployee { get; set; }
    public IFormFile? ProfilePhotoFile { get; set; } = default!;
}

File: UI/wood_track_mobile/lib/models/user_upsert_model.dart
class UserUpsertModel {
  final String firstName;
  final String lastName;
  final String email;
  final String phoneNumber;
  final DateTime? birthDate;
  final String? profilePhoto;
  final String address;
  final String? newPassword;
  final String? oldPassword;
  final int gender;

  UserUpsertModel({
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.phoneNumber,
    required this.gender,
    this.birthDate,
    this.profilePhoto,
    required this.address,
    this.newPassword,
    this.oldPassword,
  });

  factory UserUpsertModel.fromJson(Map<String, dynamic> json) {
    return UserUpsertModel(
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      phoneNumber: json['phoneNumber'],
      birthDate: json['birthDate'] != null ? DateTime.parse(json['birthDate']) : null,
      profilePhoto: json['profilePhoto'],
      address: json['address'],
      newPassword: json['newPassword'],
      gender: json['gender'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'phoneNumber': phoneNumber,
      'birthDate': birthDate?.toIso8601String(),
      'profilePhoto': profilePhoto,
      'address': address,
      'email': email,
      'newPassword': newPassword,
      'oldPassword': oldPassword,
      'gender': gender,
    };
  }
}
